### PR TITLE
Arithmetic Machine: Add equations 1-4 (Witgen not working yet)

### PR DESCRIPTION
**Pull request by georgwiese**
Depends on #985 

This PR adds the remaining pieces to implement the remaining constraints for equations 1-4, namely:
- Adds the `s` and `q0, q1, q2` witness columns
- Adds the remaining equations

The test (running `affine_256`) still works. What's still missing is:
- As expected, witgen doesn't work. I think it would suffice to add hints that compute `s`, `x3`, and `y3`. Then, witgen should be able to figure out `q0`, `q1`, and `q2`
- Some constraints needed for soundness (see TODO comment)

I also removed `arith_improved.pil`, as it doesn't anymore have any feature that the std `arith.asm` doesn't.

### CODE REVIEW

File: std/arith.asm [hunk 1]
```diff
@@ -150,44 +156,125 @@ machine Arith(CLK32_31, operation_id){
...
+    // In Polygon's version they are 19 bits, but that requires increasing the minimum degree
+    // to fit the lookup.
+    // Instead, we decompose the most significant limb into two 16-Bit limbs.
+    // Having a larger range-constraint is fine, because we're only multiplying it with 16-bit
+    // limbs of the prime, so the result is within 48 bits, still far from overflowing the
+    // Goldilocks field.
+    pol witness q0_15_high, q0_15_low, q1_15_high, q1_15_low, q2_15_high, q2_15_low;
+    q0_15_high * CLK32[0] + q0_15_low * CLK32[1] + q1_15_high * CLK32[2] + q1_15_low * CLK32[3] + q2_15_high * CLK32[4] + q2_15_low * CLK32[5] in BYTE2;
+
+    unchanged_until(q0_15_high, CLK32[31]);
```
Comment (chriseth) [inline std/arith.asm:173]: Maybe it makes sense to have a special function `|x| unchanged_until(x, CLK32[31])`?

Comment (georgwiese) [reply std/arith.asm]: Done.

File: pipeline/tests/powdr_std.rs [hunk 2]
```diff
@@ -29,6 +29,7 @@ fn split_gl_test() {
 }
 
 #[test]
+#[ignore = "Too slow"]
```
Comment (georgwiese) [inline pipeline/tests/powdr_std.rs:32]: CI failed now that we have more witness columns :/

**Comment by georgwiese** [comment]
@chriseth I just checked the diff between `test_data/pil/arith_improved.pil` and `std/arith.asm`, and I think `arith_improved` could be deleted in this PR! It doesn't have any features the std version doesn't. What do you think?

**Comment by chriseth** [comment]
Yes, please remove it!

**Review by chriseth — APPROVED** [review]

**Pull request merged**